packages/core/src/StateMachine.ts

```diff
@@ -83,7 +84,8 @@ export class StateMachine<
     TGuard,
     TDelay,
     TTag
-  >
+  >,
+  TConfig extends StateConfig2 = any
```

Comment:
We can't infer it here at the moment so it shouldn't even be added to the type parameters list. If we stop passing `TConfig` down (like suggested in another comment) then we could just pass down directly `StateConfig2` (or similar) as `TStateValue`.